sqlglot/parser.py

```diff
@@ -1912,8 +1912,16 @@ def _parse_select(
 
             this = self._parse_query_modifiers(this)
         elif (table or nested) and self._match(TokenType.L_PAREN):
-            this = self._parse_table() if table else self._parse_select(nested=True)
-            this = self._parse_set_operations(self._parse_query_modifiers(this))
+            if self._match(TokenType.PIVOT):
```

Comment:
what about select * from (pivot)

Reply:
this is a syntax error in both postgres and bigquery

Reply:
```
georgesittas=# with pivot as (select 1) select * from (pivot);
ERROR:  syntax error at or near ")"
LINE 1: with pivot as (select 1) select * from (pivot);
                                                     ^
```
<img width="1155" alt="Screenshot 2023-06-07 at 12 59 39 AM" src="https://github.com/tobymao/sqlglot/assets/46752250/3782fb27-3c57-4faf-80c8-498b1b20a309">